test/Test/Database/LSMTree/Internal/RawPage.hs

```diff
@@ -233,127 +231,66 @@ prop_ops (PageContentFits kops) =
-        Delete                 -> Entry.Delete
+    foldr1 (.&&.)
+      [     rawPageLookup rawpage (toSerialisedKey k)
+        === LookupEntry (toEntry op)
+      | (k,op) <- kops ]
   where
     rawpage = fst $ toRawPage (PageContentFits kops)
 
-prop_entries_all :: PageContentOrdered -> BS.ByteString -> Property
-prop_entries_all (PageContentOrdered kops) bs =
-    length kops /= 1 ==> rawPageLookup rawpage k === expected
+prop_entries_all :: PageContentOrdered -> Key -> Property
+prop_entries_all (PageContentOrdered kops) k =
+    length kops /= 1 ==>
+        rawPageLookup rawpage (toSerialisedKey k)
+    === maybe LookupEntryNotPresent
+              (LookupEntry . toEntry)
+              (lookup k kops)
   where
-    k = SerialisedKey $ RB.fromByteString bs
     rawpage = fst $ toRawPage (PageContentFits kops)
 
-    expected :: RawPageLookup (Entry.Entry SerialisedValue BlobSpan)
-    expected = case lookup (Key bs) kops of
-        Nothing                                    -> LookupEntryNotPresent
-        Just (Insert (Value v) Nothing)            -> LookupEntry (Entry.Insert (SerialisedValue $ RB.fromByteString v))
-        Just (Insert (Value v) (Just (BlobRef x y))) -> LookupEntry (Entry.InsertWithBlob (SerialisedValue $ RB.fromByteString v) (BlobSpan x y))
-        Just (Mupsert (Value v))                 -> LookupEntry (Entry.Mupdate (SerialisedValue $ RB.fromByteString v))
-        Just  Delete                             -> LookupEntry Entry.Delete
-
 prop_big_insert :: Key -> Maybe BlobRef -> Property
 prop_big_insert k mblobref =
-    case rawPageLookup rawpage k' of
-      LookupEntryOverflow entry suffixlen ->
-        equivalentOpEntry (Insert v mblobref)
-                          entry overflowPagesBS (fromIntegral suffixlen)
-
-      other -> counterexample (show other) False
+    rawPageLookup rawpage (toSerialisedKey k)
+   ===
+    let (prefixlen, suffixlen) =
+           fromMaybe
+            (error "expected overflow pages")
+            (pageOverflowPrefixSuffixLen =<< encodePage DiskPage4k kops)
+     in LookupEntryOverflow (toEntryPrefix op prefixlen)
+                            (fromIntegral suffixlen)
   where
-    page = PageContentFits [(k, Insert v mblobref)]
-    (rawpage, overflowPages) = toRawPage page
-    overflowPagesBS = overflowPagesToByteString overflowPages
-    k' = SerialisedKey $ RB.fromByteString (unKey k)
-
-    -- original value
-    v = Value (BS.replicate 5000 42)
-
-prop_single_entry :: Key -> Operation -> Property
-prop_single_entry k op = label (show $ null overflowPages) $
-    case rawPageLookup rawpage k' of
-      LookupEntryNotPresent ->
-        counterexample "LookupEntryNotPresent" False
-
-      LookupEntry e ->
-        equivalentOpEntry op
-                          e overflowPagesBS 0
-
-      LookupEntryOverflow e suffixlen ->
-        equivalentOpEntry op
-                          e overflowPagesBS (fromIntegral suffixlen)
+    v       = Value (BS.replicate 5000 42)
```

Comment:
I _think_ this comment referred to patches in #229. The `prop_big_insert` does still use `Value (BS.replicate 5000 42)`, but note that the `prop_single_entry` uses `PageContentSingle`, which covers multiple overflow pages.

Probably `prop_big_insert` can be retired since the case it's covering is also covered by `prop_single_entry`, but I wasn't feeling so radical.